File: 이정선/11월1주차/택배상자.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> order) {
    int answer = 0;
    stack<int> s;
    
    int index = 0;
    int orderNum = 1;
    while(order.size() >= orderNum){
       if(order[index] == orderNum) {
           answer++;
           index++;
           orderNum++;
       } else if (!s.empty() && order[index] == s.top()){
           s.pop();
           answer++;
           index++;
       } else {
           s.push(orderNum);
           orderNum++;
       }
        
    }
    
    while(!s.empty()) {
        if(order[index] == s.top()){
            answer++;
            s.pop();
            index++;
        } else break;
    }
    return answer;
}

File: 송하연/11월 1주차/택배상자.java
import java.util.*;

class Solution {
    public int solution(int[] order) {
        int answer = 0;

        Stack<Integer> stack = new Stack<>();
        
        int num=1;
        for(int target : order){
            //System.out.println("target" +target);
            //System.out.println("num" +num);
            
            while(num<=target){
                stack.push(num);
                num+=1;
            }
            
            if(!stack.isEmpty() && stack.peek()==target){
                stack.pop();
                answer++;
            }
            else break;

        }    
    
         return answer;
    }
       
}
